File: 0001-0500/0012/0012.integer-to-roman.js
/*
 * @lc app=leetcode id=12 lang=javascript
 *
 * [12] Integer to Roman
 */

// @lc code=start
/**
 * @param {number} num
 * @return {string}
 */
const val = [1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1]
const rom = ["M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"]

var intToRoman = function (N) {
    let result = "";
    for (let i = 0; i < val.length; i++) {
        while (N >= val[i]) {
            result += rom[i];
            N -= val[i];
        }
    }

    return result;
};
// @lc code=end

File: 0001-0500/0012/0012.integer-to-roman.cs
/*
 * @lc app=leetcode id=12 lang=csharp
 *
 * [12] Integer to Roman
 */

// @lc code=start
public class Solution {
    public string IntToRoman(int num) {
        var sb = new StringBuilder();
        var map = new Dictionary<int, string> {
            { 1000, "M" },
            { 900, "CM" },
            { 500, "D" },
            { 400, "CD" },
            { 100, "C" },
            { 90, "XC" },
            { 50, "L" },
            { 40, "XL" },
            { 10, "X" },
            { 9, "IX" },
            { 5, "V" },
            { 4, "IV" },
            { 1, "I" }
        };
        
        foreach (var pair in map) {
            while (num >= pair.Key) {
                sb.Append(pair.Value);
                num -= pair.Key;
            }
        }
        
        return sb.ToString();
    }
}
// @lc code=end

